contracts/wawaka/exchange/contracts/vetting_organization.cpp

```diff
@@ -0,0 +1,127 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "Dispatch.h"
+
+#include "KeyValue.h"
+#include "Environment.h"
+#include "Message.h"
+#include "Response.h"
+#include "StringArray.h"
+#include "Util.h"
+#include "Value.h"
+#include "WasmExtensions.h"
+
+#include "exchange_base.h"
+#include "issuer_authority_base.h"
+
+// -----------------------------------------------------------------
+// METHOD: initialize_contract
+//   contract initialization method
+//
+// JSON PARAMETERS:
+//   none
+//
+// RETURNS:
+//   true if successfully initialized
+// -----------------------------------------------------------------
+bool initialize_contract(const Environment& env, Response& rsp)
+{
+    return ww::exchange::exchange_base::initialize_contract(env, rsp);
+}
+
+// -----------------------------------------------------------------
+// METHOD: initialize
+//
+// JSON PARAMETERS:
+//   asset-type-id -- ecdsa public key for the asset type
+//
+// RETURNS:
+//   true if asset type id successfully saved
+// -----------------------------------------------------------------
+bool initialize(const Message& msg, const Environment& env, Response& rsp)
+{
+    return ww::exchange::issuer_authority_base::initialize_root_authority(msg, env, rsp);
+}
+
+// -----------------------------------------------------------------
+// METHOD: get_asset_type_identifier
+//
+// JSON PARAMETERS:
+//   none
+//
+// RETURNS:
+//   asset type id as a string
+// -----------------------------------------------------------------
+bool get_asset_type_identifier(const Message& msg, const Environment& env, Response& rsp)
+{
+    return ww::exchange::issuer_authority_base::get_asset_type_identifier(msg, env, rsp);
+}
+
+// -----------------------------------------------------------------
+// METHOD: get_verifying_key
+//
+// JSON PARAMETERS:
+//   none
+//
+// RETURNS:
+//   ecdsa verifying key
+// -----------------------------------------------------------------
+bool get_verifying_key(const Message& msg, const Environment& env, Response& rsp)
+{
+    return ww::exchange::exchange_base::get_verifying_key(msg, env, rsp);
+}
+
+// -----------------------------------------------------------------
+// METHOD: add_approved_issuer
+//
+// JSON PARAMETERS:
+//   issuer-verifying-key -- verifying key of the asset issuer
+//
+// RETURNS:
+//   true if key is successfully stored
+// -----------------------------------------------------------------
+bool add_approved_issuer(const Message& msg, const Environment& env, Response& rsp)
+{
```

Comment:
in this case it is approved ahead of time since the method is invoked by the contract owner. this just records the (out of band) decision to accept the (already approved) issuer.

Reply:
I see, thank you, I guess you say the same thing about the out of band vetting in the documentation as well. I forgot.